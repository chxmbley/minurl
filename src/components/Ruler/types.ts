/** Props for the decorative Ruler component */
export type RulerProps = {
  /** CSS class applied to the component */
  className?: string;
  /** Number of segments of the ruler. Three smaller segments will be rendered between each pair of segments. */
  segments?: number;
};
