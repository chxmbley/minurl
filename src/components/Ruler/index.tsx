import cn from 'classnames';
import { times } from 'lodash';
import { Fragment } from 'react';
import type { FC } from 'react';
import type { RulerProps } from './types';

/** Purely decorative component meant to appear like a ruler with major and minor segments */
const Ruler: FC<RulerProps> = ({ className, majorSegments = 5, minorSegments = 3 }) => (
  <div className={cn(className, 'flex items-end justify-between')}>
    {times(majorSegments, (i) => (
      <Fragment key={i}>
        <div className="h-6 border-r border-slate-500" />
        {i < majorSegments - 1 &&
          times(minorSegments, (j) => <div key={`${i}.${j}`} className="h-2 border-r border-slate-500" />)}
      </Fragment>
    ))}
  </div>
);

export default Ruler;
