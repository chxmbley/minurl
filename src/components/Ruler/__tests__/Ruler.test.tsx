import { render } from '@testing-library/react';
import Ruler from '~components/Ruler';

describe('Ruler', () => {
  it('renders a Ruler component', () => {
    const container = render(<Ruler />);
    expect(container).toMatchSnapshot();
  });

  it('renders a Ruler component with the specified number of segments', async () => {
    const container = render(<Ruler segments={10} />);
    expect(container).toMatchSnapshot();
  });

  it('adds a provided className to the component', async () => {
    const customClassName = 'my-custom-classname';
    render(<Ruler className={customClassName} />);
    expect(document.getElementsByClassName(customClassName)).toHaveLength(1);
  });
});
