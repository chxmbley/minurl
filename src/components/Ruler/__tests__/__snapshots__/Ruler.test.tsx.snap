// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Ruler component 1`] = `
<div>
  <div
    class="flex items-end justify-between"
  >
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
  </div>
</div>
`;

exports[`renders a Ruler component with the specified number of segments 1`] = `
<div>
  <div
    class="flex items-end justify-between"
  >
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-2 border-r border-slate-500"
    />
    <div
      class="h-6 border-r border-slate-500"
    />
  </div>
</div>
`;
